plugin/src/main/java/io/jenkins/plugins/analysis/core/portlets/PullRequestMonitoringPortlet.java

```diff
@@ -0,0 +1,130 @@
+
+/**
+ * A portlet that can be used for the pull-request-monitoring dashboard
+ * (https://github.com/jenkinsci/pull-request-monitoring-plugin).
+ * It renders an interactive sunburst diagram for each {@link ResultAction},
+ * which is registered at the current users {@link Run}.
+ *
+ * @author Simon Symhoven
+ */
+public class PullRequestMonitoringPortlet implements MonitorPortlet {
+
+    private final ResultAction action;
+
+    /**
+     * Creates a new {@link PullRequestMonitoringPortlet}.
+     *
+     * @param action
+     *              the corresponding {@link ResultAction}.
+     */
+    public PullRequestMonitoringPortlet(final ResultAction action) {
+        this.action = action;
+    }
+
+    @Override
+    public String getTitle() {
+        return action.getLabelProvider().getName();
+    }
+
+    @Override
+    @JavaScriptMethod
+    public String getId() {
+        return action.getResult().getId();
+    }
+
+    @Override
+    public int getPreferredWidth() {
+        return 350;
+    }
+
+    @Override
+    public int getPreferredHeight() {
+        return 350;
+    }
+
+    @Override
+    public String getIconUrl() {
+        return action.getIconFileName();
+    }
+
+    @Override
+    public Optional<String> getDetailViewUrl() {
+        return Optional.of("../" + action.getUrlName());
+    }
+
+    /**
+     * Get the json data for the sunburst diagram.
+     *
+     * @return
+     *          the data as json string.
+     */
+    public String getResultIssuesJsonModel() {
+        JsonObject sunburstData = new JsonObject();
+        sunburstData.addProperty("fixed", action.getResult().getFixedIssues().getSize());
+        sunburstData.addProperty("outstanding", action.getResult().getOutstandingIssues().getSize());
+
+        JsonObject newIssues = new JsonObject();
+        newIssues.addProperty("total", action.getResult().getNewIssues().getSize());
+        newIssues.addProperty("low", action.getResult().getNewIssues().getSizeOf(Severity.WARNING_LOW));
+        newIssues.addProperty("normal", action.getResult().getNewIssues().getSizeOf(Severity.WARNING_NORMAL));
+        newIssues.addProperty("high", action.getResult().getNewIssues().getSizeOf(Severity.WARNING_HIGH));
+        newIssues.addProperty("error", action.getResult().getNewIssues().getSizeOf(Severity.ERROR));
+
+        sunburstData.add("new", newIssues);
+
+        return sunburstData.toString();
+    }
+
+    /**
+     * Check if {@link AnalysisResult} has issues to display in sunburst diagram.
+     *
+     * @return
+     *          true if result has issues, else false.
+     */
+    public boolean hasIssues() {
+        return action.getResult().getFixedIssues().getSize() != 0
+                || action.getResult().getNewIssues().getSize() != 0
+                || action.getResult().getOutstandingIssues().getSize() != 0;
+    }
+
+    /**
+     * The factory for the {@link PullRequestMonitoringPortlet}.
+     */
+    @Extension
```

Comment:
If the dependency is marked as optional we need to mark the extension as optional as well
```suggestion
    @Extension(optional = true)
```